activestorage/lib/active_storage/downloader.rb

```diff
@@ -9,26 +9,22 @@ def initialize(service)
     end
 
     def open(key, checksum: nil, verify: true, name: "ActiveStorage-", tmpdir: nil)
-      open_tempfile(name, tmpdir) do |file|
-        download key, file
+      if block_given?
+        Tempfile.create(name, tmpdir, binmode: true) do |file|
```

Comment:
I'm not sure but do `Tempfile.create(&block)` and `Tempfile.open` behave differently?

I think there is a way to write this and not change that, which might be worth investigating.

Reply:
Sorry for the late response. Yes there is a slight difference between the two ([see docs about finalizer](https://github.com/ruby/tempfile/blob/86aac8890016b4ec073d63998caebb491b930feb/lib/tempfile.rb#L399-L434)). 

I've changed it back to use `open` for now and rebased.